test/Rx/Functional/Operator/GroupByTest.php

```diff
@@ -187,6 +187,79 @@ function() { throw new Exception(''); }
 
     /**
      * @test
+     * @expectedException \InvalidArgumentException
+     */
+    public function it_throws_if_keyselector_is_not_callable()
+    {
+        $xs             = $this->createHotObservableWithData(true);
+
+        $this->scheduler->startWithCreate(function() use ($xs) {
+            return $xs->groupByUntil("non-callable",
+                null,
+                function() { throw new Exception(''); }
+            )->select(function(GroupedObservable $observable) {
+                return $observable->getKey();
+            });
+        });
+    }
+
+    /**
+     * @test
+     * @expectedException \InvalidArgumentException
+     */
+    public function it_throws_if_element_selector_is_not_callable()
```

Comment:
This is a duplicate test. See: https://github.com/asm89/Rx.PHP/blob/c923ad7b9430b4a4197d1f93c7d637e19cf3e1ca/test/Rx/Functional/Operator/GroupByTest.php#L297

You should also be able to simplify all the test methods themselves.